brayns/engine/LightManager.h

```diff
@@ -0,0 +1,131 @@
+     * @param id ID of Light source to be removed
+     */
+    void removeLight(const size_t id);
+
+    /**
+     * @brief removeLight Removes a light source from the scene for a given
+     * light
+     * @param light Light source to be removed
+     */
+    void removeLight(LightPtr light);
+
+    /**
+     * @brief getLight Gets a light source from the scene for a given id.
+     * Note: If changing the light then call markModified to propagate the
+     * changes.
+     * @param id Id of the light
+     * @return Pointer to the light, nullptr if not found
+     */
+    LightPtr getLight(const size_t id);
+
+    /**
+     * @brief getLights Gets the lights
+     * @return Immutable list of all light sources
+     */
+    const std::vector<LightPtr>& getLights() const;
+
+    /**
+     * @brief clearLights Removes all light sources
+     */
+    void clearLights();
+
+    /**
+     * @brief addDirectionalLight Creates a light that shines from a specific
+     * direction not from a specific position. This light will behave as though
+     * it is infinitely far away and the rays produced from it are all parallel
+     * @param direction Light source direction
+     * @param color Light source RGB color
+     * @param intensity Amount of light emitted
+     */
+    LightPtr addDirectionalLight(const Vector3f& direction,
+                                 const Vector3f& color, float intensity);
+
+    /**
+     * @brief addSphereLight Createas a spehre light. The sphere light (or the
+     * special case point light) is a light emitting uniformly in all
+     * directions.
+     * @param position Light source position
+     * @param radius The size of the sphere light
+     * @param color Light source RGB color
+     * @param intensity Amount of light emitted
+     */
+    LightPtr addSphereLight(const Vector3f& position, float radius,
+                            const Vector3f& color, float intensity);
+
+    /**
+     * @brief addQuadLight Creates a quad light. The quad light is a planar,
+     * procedural area light source emitting uniformly on one side into the
+     * half-space.
+     * @param position World-space position of one vertex of the quad light
+     * @param edge1 Vector to one adjacent vertex
+     * @param edge2 Vector to the other adjacent vertex
+     * @param color Light source RGB color
+     * @param radius The size of the sphere light
+     */
+    LightPtr addQuadLight(const Vector3f& position, const Vector3f& edge1,
+                          const Vector3f& edge2, const Vector3f& color,
+                          float intensity);
+
+    /**
+     * @brief addSpotLight Creates a spot light. The spotlight is a light
+     * emitting into a cone of directions.
+     * @param position The center of the spotlight, in world-space
+     * @param direction  Main emission direction of the spot
+     * @param openingAngle  Full opening angle (in degree) of the spot; outside
```

Comment:
Is it better to use degrees than radians?

Reply:
I leave it like this to mirror OSPRay properties.